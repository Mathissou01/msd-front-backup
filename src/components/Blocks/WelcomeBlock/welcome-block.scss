@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/01-settings/settings.layout" as layout;
@use "src/styles/02-tools/tools.include-media" as im;
@use "src/styles/02-tools/tools.typography" as typo;
@use "src/styles/02-tools/tools.z-index" as index;

.c-WelcomeBlock {
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: flex-start;

  &__Heading {
    z-index: index.z("site-content");
    display: flex;
    flex-direction: column;
    gap: 8px;

    @include im.media("<laptop") {
      min-height: 0;
      margin: 0;
    }

    @include im.media(">=laptop") {
      width: 68.86%;
      max-width: 850px;
      min-height: 186px;
      margin-top: 60px;
      margin-right: 0;
      margin-left: 60px;
    }
  }

  &__Title {
    @include typo.text("huge");

    color: colors.$expert-blue;
  }

  &__Subtitle {
    @include typo.text("body");
  }

  &__SvgContainer {
    z-index: index.z("background");
    width: 100%;
    min-height: calc(
      // Avoid CLS with expected aspect-ratio
      100vw / (320 / 200)
    );
    pointer-events: none;

    @include im.media(">=mobile", "<laptop") {
      min-height: calc(
        // Avoid CLS with expected aspect-ratio
        100vw / (768 / 280)
      );
      margin-top: -4px;
    }

    @include im.media(">=laptop") {
      min-height: calc(
        // Avoid CLS with expected aspect-ratio
        (100vw - #{layout.$sidebar-width}) / (1233 / 500)
      );
      margin-top: -242px;
    }

    svg {
      width: 100%;
      height: 100%;

      #primary-color {
        fill: colors.$brand-primary;
      }

      #secondary-color {
        fill: colors.$brand-secondary;
      }
    }
  }

  &__Svg {
    &_mobile {
      display: initial;

      @include im.media(">=mobile") {
        display: none;
      }
    }

    &_tablet {
      display: none;

      @include im.media(">=mobile", "<laptop") {
        display: initial;
      }
    }

    &_desktop {
      display: none;

      @include im.media(">=laptop") {
        display: initial;
      }
    }
  }
}
