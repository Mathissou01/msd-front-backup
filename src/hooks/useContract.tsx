import React, { useContext } from "react";
import { ContractEntity } from "../graphql/codegen/generated-types";

export interface IContractContext {
  contract: ContractEntity;
  contractId: `${number}`;
  colors: { [key: string]: string };
}

export const ContractContext = React.createContext<IContractContext>({
  contract: {},
  contractId: "0",
  colors: {},
});

export const useContract = () => useContext(ContractContext);
