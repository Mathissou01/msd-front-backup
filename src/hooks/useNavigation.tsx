import React, { useContext } from "react";

export enum ENavigationPages {
  "HomeLink" = "/",
  // Editorial
  "ComponentLinksNews" = "/actualites-evenements",
  "ComponentLinksQuizzes" = "/quiz",
  "ComponentLinksTips" = "/astuces",
  "ComponentLinksFrees" = "/contenu-libre", // Path is concatenated with name as normalized string
  "ComponentLinksContactUs" = "/contact",
  "ComponentLinksMyWasteCounter" = "/mon-compteur-dechets",
  // Services
  "ComponentLinksRecyclingGuide" = "/service/guide-tri",
  "ComponentLinksDropOffMap" = "/service/carte",
  "ComponentLinksPickUpDay" = "/service/collecte-adresse",
  "ComponentLinksRequest" = "/service/demandes",
  // Misc
  "ComponentLinksExternal" = "/", // TODO: handle external links
}

export interface INavigationContext {
  currentPage: ENavigationPages | string;
  setCurrentPage: (pageName: ENavigationPages | string) => void;
}

export const NavigationContext = React.createContext<INavigationContext>({
  currentPage: ENavigationPages.HomeLink,
  setCurrentPage: () => null,
});

export const useNavigation = () => useContext(NavigationContext);
