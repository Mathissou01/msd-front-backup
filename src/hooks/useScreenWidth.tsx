import React, { useState, useEffect, useMemo, useContext } from "react";

const isWindow = typeof window !== "undefined";
const getRawDesktopBreakpoint = () =>
  isWindow
    ? Number.parseInt(
        getComputedStyle(document.documentElement).getPropertyValue(
          "--breakpoint-laptop",
        ),
      )
    : null;
const getDesktopBreakpoint = () => {
  const rawDesktopBreakpoint = getRawDesktopBreakpoint();
  return rawDesktopBreakpoint !== null && !isNaN(rawDesktopBreakpoint)
    ? rawDesktopBreakpoint
    : 1440;
};
const getIsDesktop = (width: number | undefined) => {
  const desktopBreakpoint = getDesktopBreakpoint();
  return width && desktopBreakpoint ? width >= desktopBreakpoint : undefined;
};

export interface IScreenWidth {
  windowWidth?: number;
  isDesktop?: boolean;
}

export default function useScreenWidth(): IScreenWidth {
  const getWidth = (): number | undefined =>
    isWindow ? window.innerWidth : undefined;
  const [windowWidth, setWindowWidth] = useState<number | undefined>(
    getWidth(),
  );
  const [isDesktop, setIsDesktop] = useState<boolean | undefined>(undefined);

  useEffect(() => {
    if (isWindow) {
      const resize = () => {
        const width = getWidth();
        setWindowWidth(width);
        setIsDesktop(getIsDesktop(width));
      };
      resize();
      window.addEventListener("resize", resize);
      return () => window.removeEventListener("resize", resize);
    }
  }, []);

  return useMemo(() => ({ windowWidth, isDesktop }), [windowWidth, isDesktop]);
}

export const IsDesktopContext = React.createContext<boolean | undefined>(
  undefined,
);

export const useIsDesktopContext = () => useContext(IsDesktopContext);
