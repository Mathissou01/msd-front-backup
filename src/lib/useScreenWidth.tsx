import { useState, useEffect } from "react";

export default function useScreenWidth() {
  const [windowWidth, setWindowWidth] = useState<number | null>(null);
  const [isDesktop, setIsDesktop] = useState<boolean>(false);
  const isWindow = typeof window !== "undefined";
  const getWidth = () => (isWindow ? window.innerWidth : windowWidth);
  const getDesktopBreakpoint = () =>
    isWindow
      ? Number.parseInt(
          getComputedStyle(document.documentElement).getPropertyValue(
            "--breakpoint-desktop",
          ),
        )
      : null;
  const resize = () => {
    const width = getWidth();
    const rawDesktopBreakpoint = getDesktopBreakpoint();
    // Set a default value of 1440 for the desktop breakpoint for Jest
    const desktopBreakpoint =
      rawDesktopBreakpoint !== null && !isNaN(rawDesktopBreakpoint)
        ? rawDesktopBreakpoint
        : 1440;
    setWindowWidth(width);
    setIsDesktop(
      width && desktopBreakpoint ? width >= desktopBreakpoint : false,
    );
  };

  useEffect(() => {
    if (isWindow) {
      resize();
      window.addEventListener("resize", resize);
      return () => window.removeEventListener("resize", resize);
    }
    //eslint-disable-next-line
  }, [isWindow]);

  return { windowWidth, isDesktop };
}
