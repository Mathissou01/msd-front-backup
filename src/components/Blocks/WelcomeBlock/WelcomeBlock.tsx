import HeroIlluDesktop from "public/images/Hero-illu-desktop.svg";
import HeroIlluTablet from "public/images/Hero-illu-tablet.svg";
import HeroIlluMobile from "public/images/Hero-illu-mobile.svg";
import "./welcome-block.scss";

interface IWelcomeBlockProps {
  title: string;
  subtitle: string;
}

export default function WelcomeBlock({ title, subtitle }: IWelcomeBlockProps) {
  return (
    <section className="c-WelcomeBlock">
      <div className="c-WelcomeBlock__Heading">
        <h1 className="c-WelcomeBlock__Title">{title}</h1>
        <span className="c-WelcomeBlock__Subtitle">{subtitle}</span>
      </div>
      <div className="c-WelcomeBlock__SvgContainer">
        <HeroIlluDesktop
          className="c-WelcomeBlock__Svg_desktop"
          data-testid="hero-illu-desktop"
        />
        <HeroIlluTablet
          className="c-WelcomeBlock__Svg_tablet"
          data-testid="hero-illu-tablet"
        />
        <HeroIlluMobile
          className="c-WelcomeBlock__Svg_mobile"
          data-testid="hero-illu-mobile"
        />
      </div>
    </section>
  );
}
