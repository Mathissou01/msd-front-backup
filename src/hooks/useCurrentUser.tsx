import React, {
  Dispatch,
  SetStateAction,
  createContext,
  useContext,
  useEffect,
  useState,
} from "react";
import { User } from "../lib/user";

interface UserContextValues {
  currentUser: User | null;
  setCurrentUser: Dispatch<SetStateAction<User | null>>;
  login: () => void;
  logout: () => void;
}

export const useCurrentUser = (): UserContextValues => {
  const context = useContext(UserContext);
  if (!context) {
    throw new Error();
  }
  return context;
};

const UserContext = createContext<UserContextValues>({} as UserContextValues);

export const UserProvider = ({ children }: { children: React.ReactNode }) => {
  const [currentUser, setCurrentUser] = useState<User | null>(null);

  const getUser = async () => {
    try {
      const storedUser = localStorage.getItem("user");
      if (storedUser && storedUser !== "null") {
        setCurrentUser(JSON.parse(storedUser));
      }
    } catch (error) {
      console.error(error);
    }
  };

  const login = () => {
    const user = {
      id: "1",
      email: "john.doe@gmail.com",
      firstname: "John",
      lastname: "Doe",
      activeCounter: true,
      householdSize: 1,
      userType: "particular",
      dwellingType: "house",
      addressLabel: "1 rue de la paix, 75000 Paris",
    };
    setCurrentUser(user);
    localStorage.setItem("user", JSON.stringify(user));
  };

  const logout = () => {
    setCurrentUser(null);
    localStorage.removeItem("user");
  };

  useEffect(() => {
    getUser();
  }, []);

  useEffect(() => {
    localStorage.setItem("user", JSON.stringify(currentUser));
  }, [currentUser]);

  const contextValue: UserContextValues = {
    currentUser,
    setCurrentUser,
    login,
    logout,
  };

  return (
    <UserContext.Provider value={contextValue}>{children}</UserContext.Provider>
  );
};

export default UserContext;
