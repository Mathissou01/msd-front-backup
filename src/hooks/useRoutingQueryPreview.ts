import { useRouter } from "next/router";
import { TPreviewTypes } from "../lib/preview";

export function useRoutingQueryPreview<T extends string | TPreviewTypes>(
  queryParam: string,
  validateQueryParam: (param: string) => boolean,
): T | undefined {
  const router = useRouter();
  const query = router.query[queryParam];
  if (query?.toString()) {
    if (!validateQueryParam(query.toString())) {
      void router.push("/404");
      return undefined;
    }
    return query?.toString() as T;
  } else if (typeof window === undefined) {
    void router.push("/404");
  }

  return undefined;
}
