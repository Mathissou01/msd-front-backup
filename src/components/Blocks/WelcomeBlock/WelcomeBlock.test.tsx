import { render, screen } from "@testing-library/react";
import { IsDesktopContext } from "../../../hooks/useScreenWidth";
import WelcomeBlock from "./WelcomeBlock";

describe("WelcomeBlock", () => {
  it("renders in mobile mode", () => {
    const { container } = render(
      <IsDesktopContext.Provider value={false}>
        <WelcomeBlock />
      </IsDesktopContext.Provider>,
    );

    const heroMobile = screen.getByTestId("hero-illu-mobile");
    const heroDesktop = screen.queryByTestId("hero-illu-desktop");
    expect(heroMobile).toBeInTheDocument();
    expect(heroDesktop).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renders in desktop mode", () => {
    const { container } = render(
      <IsDesktopContext.Provider value={true}>
        <WelcomeBlock />
      </IsDesktopContext.Provider>,
    );

    const heroDesktop = screen.getByTestId("hero-illu-desktop");
    const heroMobile = screen.queryByTestId("hero-illu-mobile");
    expect(heroDesktop).toBeInTheDocument();
    expect(heroMobile).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
