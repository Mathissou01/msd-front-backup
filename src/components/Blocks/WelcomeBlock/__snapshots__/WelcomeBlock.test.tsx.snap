// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeBlock renders in desktop mode 1`] = `
<div>
  <section
    class="c-WelcomeBlock"
  >
    <div
      class="c-WelcomeBlock__Heading"
    >
      <h1
        class="c-WelcomeBlock__Title"
      />
      <span
        class="c-WelcomeBlock__Subtitle"
      />
    </div>
    <div
      class="c-WelcomeBlock__SvgContainer"
    >
      <div
        class="c-WelcomeBlock__Svg_desktop"
        data-testid="hero-illu-desktop"
      />
    </div>
  </section>
</div>
`;

exports[`WelcomeBlock renders in mobile mode 1`] = `
<div>
  <section
    class="c-WelcomeBlock"
  >
    <div
      class="c-WelcomeBlock__Heading"
    >
      <h1
        class="c-WelcomeBlock__Title"
      />
      <span
        class="c-WelcomeBlock__Subtitle"
      />
    </div>
    <div
      class="c-WelcomeBlock__SvgContainer"
    >
      <div
        class="c-WelcomeBlock__Svg_mobile"
        data-testid="hero-illu-mobile"
      />
    </div>
  </section>
</div>
`;
